tests/test_ebay_search.py

```diff
@@ -21,22 +21,30 @@ def test_get_json(self):
 				# pp = pprint.PrettyPrinter(indent=2)
 				# pp.pprint(data)
 
-    # TODO fix universal pagination
-		@unittest.skipIf((os.getenv("DEBUGAPI_KEY") == None), "no api_key provided")
+		@unittest.skipIf((os.getenv("API_KEY") == None), "no api_key provided")
```

Comment:
@dimitryzub here's another example test case for your test scenarios.

Reply:
@ilyazub Currently, I'm not really sure what else to add here. 

The issue I had with eBay is I had to check if the current page == next page, and if equal - exit the loop i.e `raise StopIteration`, which is done by checking:

https://github.com/serpapi/google-search-results-python/blob/b733a1a4375c8393e8242ae745f9b6c75c0d42f6/serpapi/pagination.py#L57-L58

If I'm not getting it wrong, it should work correctly.